### PR TITLE
[Low-code CDK] Add ability to remove fields

### LINKED ISSUE #12988: Low-code connectors: Support removing fields from records
## What
The freshsales and twilio connectors remove fields from the records before emitting them. This should be an option in low-code cdk

## How
```
extractor:
  class_name: airbyte_cdk.sources.declarative.extractors.jq.JqFetcher
    decoder: "*ref(decoder)"
    transformer:
      class_name: airbyte_cdk.sources.declarative.extractors.FieldTranformer
        transform:
          remove_fields:
           - field
```

removing the fields can then be done by iterating over the records and over the fields to remove
```
for record in records:
  for field in self._fields_to_remove:
    record.pop(field, None)
yield from records
```

## Alternative
The simplest, but not so low-code way to do this is with jello with a non-trivial transform
`"[{k:v for k,v in d.items() if k != 'value_to_remove'} for d in _.data]"`

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Closes #12988 

The main differences from the issue as spec'd by Alex:
1. Transformations are defined on the Stream level rather than on the retriever or extractor. The reason is that transformations are probably going to be stream-specific rather than retrieval-strategy specific. Open to feedback on this though!
2. Transformations are defined as an ordered list then applied to the records in the definition order. The main downside is that this is not "declarative". The main upside is that we push the concern of dependencies between transformations up to the developer. Otherwise there would be some weird undefined behavior (what if I remove a field then add a field with the same name, or add a field that depends on a field that is removed?). 

## Recommended reading order
1. `DeclarativeStream`
3. `RecordTransformation`
4. `RemoveFieldTransformation`
5. tests

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self._field_pointers = field_pointers
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        :param record: The record to be transformed
+        :return: the input record with the requested fields removed
+        """
+        for pointer in self._field_pointers:
+            try:
+                dpath.util.delete(record, pointer)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py:30]: very cool! can we make sure there's a test for removing nested fields?

Comment (sherifnada) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py:50]: done

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def __init__(
         elif isinstance(request_options_provider, dict):
             request_options_provider = InterpolatedRequestOptionsProvider(config=config, **request_options_provider)
         self._name = name
-        self._authenticator = authenticator
+        self._authenticator = authenticator or NoAuth
```
[the remark this answers is not in the document]
Comment (sherifnada) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py]: fixed

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+from airbyte_cdk.sources.declarative.types import FieldPointer
+
+
+class RemoveFields(RecordTransformation):
+    """
+    A transformation which removes fields from a record. The fields removed are designated using FieldPointers.
+    During transformation, if a field or any of its parents does not exist in the record, no error is thrown.
+
+    If an input field pointer references an item in a list (e.g: ["k", 0] in the object {"k": ["a", "b", "c"]}) then
+    the object at that index is set to None rather than being not entirely removed from the list. TODO change this behavior.
```
[the remark this answers is not in the document]
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py:19]: agreed that this is not a common use case. sounds good to me!

**Review by girarda — APPROVED** [review]

**Review by girarda — APPROVED** [review]

**Pull request merged**